dali/kernels/slice/slice_flip_normalize_permute_pad_gpu.h

```diff
@@ -45,19 +53,76 @@ class SliceFlipNormalizePermutePadGpu {
     KernelRequirements req;
     ScratchpadEstimator se;
     const size_t num_samples = in.size();
-    se.add<detail::SampleDesc<Dims>>(AllocType::Host, num_samples);
-    se.add<detail::SampleDesc<Dims>>(AllocType::GPU, num_samples);
 
-    DALI_ENFORCE(args[0].mean.size() == args[0].inv_stddev.size());
-    size_t norm_args_size = args[0].mean.size();
-    if (norm_args_size > 0) {
-      se.add<float>(AllocType::Host, 2 * norm_args_size);
-      se.add<float>(AllocType::GPU,  2 * norm_args_size);
+    norm_args_size_ = -1;
+    has_channels_ = false;
+    need_normalize_ = false;
+    for (const auto& sample_args : args) {
+      if (norm_args_size_ == -1) {
+        norm_args_size_ = sample_args.mean.size();
+        need_normalize_ = norm_args_size_ > 0;
+        has_channels_ = norm_args_size_ > 1;
+      }
+
+      if (sample_args.mean.size() != norm_args_size_ ||
+          sample_args.inv_stddev.size() != norm_args_size_)
```

Comment:
FYI: `Normalize` operator allows one of them to be a scalar and broadcasts it to match the other.